### PR TITLE
Updating worker sdk Extensions csproj generator to support v4

### LINKED ISSUE #617: Update extensions csproj generator to support Functions 4.0
We need to update the generator to remove the dependency on .NET Core 3.1 in Functions 4.0 scenarios.

### PR DESCRIPTION
**Pull request by soninaren**
closes #617

### CODE REVIEW

File: sdk/Sdk/ExtensionsCsprojGenerator.cs [hunk 1]
```diff
@@ -49,14 +49,21 @@ private void WriteExtensionsCsProj(string filePath)
             File.WriteAllText(filePath, csprojContent);
         }
 
+
```
Comment (liliankasem) [inline sdk/Sdk/ExtensionsCsprojGenerator.cs:52]: nit: were the extra lines here by accident?

File: sdk/Sdk/ExtensionsCsprojGenerator.cs [hunk 2]
```diff
@@ -49,14 +49,21 @@ private void WriteExtensionsCsProj(string filePath)
...
 
+            string targetFrameWork = GetTargetFrameWork();
+
+            string netSdkVersion = GetNetSdkVersion();
+
             return $@"
 <Project Sdk=""Microsoft.NET.Sdk"">
     <PropertyGroup>
-        <TargetFramework>{_targetFramework}</TargetFramework>
+        <TargetFramework>{targetFrameWork}</TargetFramework>
```
Comment (liliankasem) [inline sdk/Sdk/ExtensionsCsprojGenerator.cs:66]: I think framework is one word so the original camelcase of the variable makes more sense, or is this something out of our control?

File: sdk/Sdk/ExtensionsCsprojGenerator.cs [hunk 3]
```diff
@@ -69,13 +76,33 @@ internal string GetCsProjContent()
...
+        {
+            if (_azureFunctionsVersion.StartsWith("v3", StringComparison.OrdinalIgnoreCase))
+            {
+                return "3.0.11";
+            }
+
+            return "4.0.0-preview2";
+        }
+
+        private string GetTargetFrameWork()
```
Comment (liliankasem) [inline sdk/Sdk/ExtensionsCsprojGenerator.cs:96]: nit: same here about using `Framework` over `FrameWork`

File: sdk/Sdk/ExtensionsCsprojGenerator.cs [hunk 4]
```diff
@@ -69,13 +76,33 @@ internal string GetCsProjContent()
...
+        {
+            if (_azureFunctionsVersion.StartsWith("v3", StringComparison.OrdinalIgnoreCase))
+            {
+                return "netcoreapp3.1";
+            }
+
+            return "net6.0";
```
Comment (liliankasem) [inline sdk/Sdk/ExtensionsCsprojGenerator.cs:103]: Just wondering, would it be worth us using constant strings for these version variables so they're defined in one place or nah?

File: sdk/Sdk/ExtensionsCsprojGenerator.cs (hunk 1 above)
Comment (soninaren) [reply sdk/Sdk/ExtensionsCsprojGenerator.cs:52]: yeah, let me fix that.

File: sdk/Sdk/ExtensionsCsprojGenerator.cs (hunk 4 above)
Comment (soninaren) [reply sdk/Sdk/ExtensionsCsprojGenerator.cs:103]: Should definitely use a constant. Will change it

**Review by liliankasem — APPROVED** [review]

**Pull request merged**